### PR TITLE
Disassembly: add arm32 support

### PR DESCRIPTION
**Pull request by xxr0ss**
- Simplified arm64 instructon dissection
- Removed hard-coded arch strings
- Tests

solved issue https://github.com/angr/angr/issues/3485, PR https://github.com/angr/angr/pull/3621

### CODE REVIEW

**Comment by xxr0ss** [comment]
I'll rebase this branch when PR https://github.com/angr/angr/pull/3750 is merged

File: tests/test_disassembly.py [hunk 1]
```diff
@@ -1,21 +1,22 @@
...
 
+import archinfo
 import angr
 from angr.analyses import Disassembly
 from angr.analyses.disassembly import MemoryOperand, Instruction
 
 
 class TestDisassembly(TestCase):
-    def test_arm64_disect_instruction(self):
+    def test_arm_dissect_instructions(self):
```
Comment (ltfish) [inline tests/test_disassembly.py:11]: Why are you changing `arm64` to `arm`?

File: tests/test_disassembly.py [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
         proj = angr.load_shellcode(
             b"\x00\xe4\x00\x6f"
             b"\x43\x3c\x0b\x0e"
             b"\x54\x9a\xb7\x72"
             b"\xfc\x6f\xba\xa9"
             b"\x88\x03\x98\x1a"
             b"\x00\x60\x01\x4e",
-            "AARCH64",
+            archinfo.ArchAArch64(),
```
Comment (ltfish) [inline tests/test_disassembly.py:19]: IMO this change is not necessary. angr (in fact, archinfo) should be able to convert "AArch64" to `archinfo.ArchAArch64`.

File: tests/test_disassembly.py [hunk 3]
```diff
@@ -38,6 +39,25 @@ def test_arm64_disect_instruction(self):
...
+
+        block = proj.factory.block(0)
+        disasm = proj.analyses[Disassembly].prep()(ranges=[(block.addr, block.addr + block.size)])
+        insns = [r for r in disasm.raw_result if isinstance(r, Instruction)]
+        rendered_insns = [i.render()[0].lower() for i in insns]
+        assert all(
+            i in rendered_insns[0] for i in ('{', '}', 'lr', 'pc')
+        )
+        assert 'sb!' in rendered_insns[1]
+        assert rendered_insns[1].endswith('^')
```
Comment (ltfish) [inline tests/test_disassembly.py:59]: Thanks for adding an ARM-specific test case. Can you please separate the AArch64 test case and this ARM test case into two methods?

File: tests/test_disassembly.py [hunk 4]
```diff
@@ -48,7 +68,7 @@ def test_mips32_missing_offset_in_instructions(self):
             b"\x24\xA5\x5E\x38"
             b"\x03\x20\xF8\x09"
             b"\x24\x04\x00\x02",
-            "MIPS32",
+            archinfo.ArchMIPS32(),
```
Comment (ltfish) [inline tests/test_disassembly.py:71]: This change is unnecessary.

**Review by ltfish — CHANGES_REQUESTED** [review]
Please take a look at my comments above.

**Comment by xxr0ss** [comment]
@ltfish Thanks for reviewing my code, I'll fix these soon

**Review by xxr0ss — COMMENTED** [review]
Done

**Review by xxr0ss — COMMENTED** [review]
Ready to merge

**Review by ltfish — APPROVED** [review]

**Comment by ltfish** [comment]
Thank you!

**Comment by xxr0ss** [comment]
Happy to help! @ltfish BTW https://github.com/angr/angr/issues/3485, https://github.com/angr/angr/pull/3621 can be closed

**Pull request merged**